package_manager/dpkg_parser.py

```diff
@@ -0,0 +1,99 @@
+import argparse
+import gzip
+import re
+import io
+import urllib2
+import json
+
+PACKAGES_FILE_NAME = "file/Packages.json"
+DEB_FILE_NAME = "pkg.deb"
+
+parser = argparse.ArgumentParser(
+    description = "Downloads a deb package from a package source file"
+)
+
+parser.add_argument("--packages-file", action='store',
+                    help='The file path of the Packages.gz file')
+parser.add_argument("--package-name", action='store',
+                    help='The name of the package to search for and download')
+
+parser.add_argument("--download-and-extract-only", action='store',
+                    help='If True, download Packages.gz and make urls absolute from mirror url')
+parser.add_argument("--mirror-url", action='store',
+                    help='The base url for the package list mirror')
+parser.add_argument("--arch", action='store',
+                    help='The target architecture for the package list')
+parser.add_argument("--distro", action='store',
+                    help='The target distribution for the package list')
+
+def main():
+    args = parser.parse_args()
+    if args.download_and_extract_only:
+        download_package_list(args.mirror_url, args.distro, args.arch)
+    else:
+        download_dpkg(args.packages_file, args.package_name)
+
+
+def download_dpkg(packages_file, package_name):
+    with open(packages_file, 'rb') as f:
+        metadata = json.load(f)
+    for pkg in metadata:
+        if pkg["Package"] == package_name:
+            print(pkg)
+            buf = urllib2.urlopen(pkg["Filename"])
+            with open(DEB_FILE_NAME, 'w') as f:
+                f.write(buf.read())
+            break
+
+def download_package_list(mirror_url, distro, arch):
+    url = "%s/debian/dists/%s/main/binary-%s/Packages.gz" % (
```

Comment:
will path.join work on windows?  Not too familiar with the python impl but can it also be used for URLs?